axelrod/history.py

```diff
@@ -0,0 +1,88 @@
+from collections import Counter
+
+from axelrod.action import Action, actions_to_str
+
+C, D = Action.C, Action.D
+
+
+class History(object):
+    """
+    History class to track the history of play and metadata including
+    the number of cooperations and defections, and if available, the
+    opponents plays and the state distribution of the history of play.
+    """
+
+    def __init__(self, plays=None, coplays=None):
+        self._plays = []
+        # Coplays is tracked mainly for computation of the state distribution
+        # when cloning or dualing.
+        self._coplays = []
+        self._actions = Counter()
+        self._state_distribution = Counter()
+        if plays:
+            self.extend(plays, coplays)
+
+    def append(self, play, coplay):
+        self._plays.append(play)
+        self._actions[play] += 1
+        if coplay:
+            self._coplays.append(coplay)
+            self._state_distribution[(play, coplay)] += 1
+
+    def copy(self):
+        return History(plays=self._plays, coplays=self._coplays)
+
+    def dual(self):
+        """Creates a dual history for use with DualTransformer."""
```

Comment:
I think we need to be careful with the dual here, it's not the same as flipping all the moves:

```
  310     The Dual of a strategy will return the exact opposite set of moves to the                                                                   
  311     original strategy when both are faced with the same history.                                                                                
  312                                                                                                                                                 
  313     A formal definition can be found in [Ashlock2010]_.                                                                                         
  314     http://doi.org/10.1109/ITW.2010.5593352    
```

I see that this test is not modified
https://github.com/Axelrod-Python/Axelrod/pull/1241/files#diff-1a0ceac0c6b9bdd7b042f6230ee3fb82R210 so I'm pretty confident it's all fine. I'm just not sure the name of the method is the best here, perhaps `flipped_plays`? The actual `dual` happens in the transformer.

Reply:
Sure, I wasn't sure what the proper name for this was.

Reply:
Yup :+1: